src/engine/strat_engine/blockdev.rs

```diff
@@ -129,10 +135,39 @@ impl BlockDev for StratBlockDev {
     fn devnode(&self) -> PathBuf {
         self.devnode.clone()
     }
+
+    fn user_info(&self) -> Option<&str> {
+        self.user_info.as_ref().map(|x| &**x)
+    }
+
+    fn set_user_info(&mut self, user_info: Option<&str>) -> () {
+        self.user_info = user_info.map(|x| x.to_owned());
+    }
+
+    fn hardware_info(&self) -> Option<&str> {
+        self.hardware_info.as_ref().map(|x| &**x)
+    }
+
+    fn initialization_time(&self) -> DateTime<Utc> {
+        Utc.timestamp(self.bda.initialization_time() as i64, 0)
+    }
+
+    fn total_size(&self) -> Sectors {
+        self.avail_range().1
+    }
+
+    fn state(&self) -> BlockDevState {
+        // TODO: Implement states for blockdevs
+        BlockDevState::InUse
+    }
 }
 
 impl Recordable<BlockDevSave> for StratBlockDev {
     fn record(&self) -> BlockDevSave {
-        BlockDevSave { devnode: self.devnode.clone() }
+        BlockDevSave {
+            dev: Some(self.devnode.clone()),
```

Comment:
If we really might not have a devnode to record, then it seems like BlockDev's devnode field should be an Option type, like the other two fields that get recorded.

Reply:
I think we will always have a devnode in StratBlockDev. I would flip it around and ask: why have devnode in BlockDevSave? It seems like it might be handy to know (maybe just informationally, to say "can't activate pool b/c blockdev [uuid] last seen at [devnode] is missing") but if it's not in the json then it shouldn't be an error because it's not used for anything critical, and maybe in the future we want to leave it out.

So this might be a spec issue worth further discussion around the value of it in the json, but as the spec is now I think this code is correct.

Reply:
I've filed an issue.